examples/eval.rs

```diff
@@ -5,29 +5,31 @@ fn main() {
 }
 
 fn app(cx: Scope) -> Element {
-    let eval = dioxus_desktop::use_eval(cx);
-    let script = use_state(cx, String::new);
-    let output = use_state(cx, String::new);
+    let mut eval = dioxus_html::prelude::use_eval(
+        cx,
+        r#"
+        dioxus.send("Hi from JS!");
+        let msg = await dioxus.recv();
+        console.log(msg);
+        "#,
+    );
 
-    cx.render(rsx! {
-        div {
-            p { "Output: {output}" }
-            input {
-                placeholder: "Enter an expression",
-                value: "{script}",
-                oninput: move |e| script.set(e.value.clone()),
-            }
-            button {
-                onclick: move |_| {
-                    to_owned![script, eval, output];
-                    cx.spawn(async move {
-                        if let Ok(res) = eval(script.to_string()).await {
-                            output.set(res.to_string());
-                        }
-                    });
-                },
-                "Execute"
-            }
+    eval.run().unwrap();
+    eval.send("Hi from Rust!".into()).unwrap();
+
+    let receiver = eval.receiver();
+
+    let future = use_future(cx, (), |_| async move { receiver.recv().await.unwrap() });
```

Comment:
Running this example on my M1 mac, it panics unwrapping at this line

Reply:
Does it say why?

Reply:
Nevermind, I'm getting it too.

Reply:
Should be fixed now.

Reply:
This seems like it became an issue again. If the eval gets dropped it will panic

This fixes the issue, but if it is possible to make the original example work, that would probably be better:

```rust
use dioxus::prelude::*;

fn main() {
    dioxus_desktop::launch(app);
}

fn app(cx: Scope) -> Element {
    let eval_provider = use_eval(cx);

    let future = use_future(cx, (), |_| {
        let mut eval = eval_provider(
            r#"
            dioxus.send("Hi from JS!");
            let msg = await dioxus.recv();
            console.log(msg);
        "#,
        );
        async move {
            eval.run().unwrap();
            eval.send("Hi from Rust!".into()).unwrap();

            let receiver = eval.receiver();
            receiver.recv().await.unwrap()
        }
    });

    match future.value() {
        Some(v) => cx.render(rsx!(
            p { "{v}" }
        )),
        _ => cx.render(rsx!(
            p { "hello" }
        )),
    }
}
```

Reply:
Realistically the error should be handled, but we could create a wrapper type around the channel that has the same methods but instead of errors it returns Option<Value>